---
title: "mRNA / micro RNA"
subtitle: Program 07
author: "Francois Collin"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
always_allow_html: true
---


## miRNA Seq - Differential expression analysis

```{r, cfg_prog}
cfg_prog <- yaml::read_yaml("_prog.yml")
```

The variation in messenger RNA (mRNA) and micro-RNA (miRNA) abundance
measured by RNA-Seq associated either with the dysglycemia and/or the
3-month exercise intervention were investigated by a Differential
Expression analysis (DE). At baseline, the DE models included only the
effect of the dysglycemia-constrast group, consistently with the Anova
model applied for demographics and baseline anthropometrics.
To do so, genetic features which were never found expressed were
discarded. Then RNA-Seq libraries were included if reaching a read depth
threshold (total number of reads per sample) fixed by visual examination
of the association between sample gene diversity and sample depth. For
the remaining libraries, genetic features were included if the count per
million reads (CPM) was greater than
`r cfg_prog$rna$cpm_threshold$allmirna` in at least
`r cfg_prog$rna$cpm_threshold$fraccol * 100`% of the samples.
The DE analysis implementation proposed by @Love2014 with the R package
`DESeq2` was used to fit the models; it is based on a negative binomial
distribution accounting for read variability, dispersion corrected after
trends seen across all samples and genes [@Duendar2018]. In addition,
the number of tested genes being high, raw p.values were adjusted
according to Benjamini and Hochberg's method also known as
False-Discovery Rate (FDR).

The analysis was realized with R [v4.2.0, @RCT2022]. Note the
use of the additional packages downloaded from the RStudio package manager
repository (freeze date 2022-05-04, [repos](
https://packagemanager.rstudio.com/all/2022-05-04+Y3JhbiwyOjQ1MjYyMTU7NTVFQjhGODU)):

- `DESeq2` for differential expression analysis and
variance stabilizing transformation [version 1.36.0, @Love2014]
- `ggplot2` for graphics [version 3.3.6, @Wickham2016]
- `MultiAssayExperiment` for management of multi-omics experiment objects
[version 1.22.0, @Ramos2017].

Target:

- [x] miRNA DE at baseline without confounding factors.
- [x] mRNA DE at baseline without confounding factors.
- [x] miRNA DE at month 3 without confounding factors.
- [x] mRNA DE at month 3 without confounding factors.
- [x] miRNA DE variation induced by the exercise intervention.
- [x] mRNA DE variation induced by the exercise intervention.

```{r, settings}
devtools::load_all("src/pkg/dbs.data")
devtools::load_all("src/pkg/latarnia.utils")

knitr::opts_chunk$set(results = cfg_prog$knitr$results)
library(assertthat)
library(ggplot2)
source("R/ngs.R")
source("R/inches.R")
```

### Materials and Methods

#### Helper functions

```{r}
# Help for data pre-processing
filter_for_depth <- function(mae, assay, min_depth) {
  mae[, colSums(mae[[ngs_assay]]) > min_depth, ]
}

filter_for_coldata <- function(mae, col = "avisit", is) {
  mae[, colData(mae)[[col]] %in% is, ]
}

filter_for_low_expr <- function(mae, assay, min_cpm,
                                frac_cols = cfg_prog$rna$cpm_threshold$fraccol
) {
  # Genes expressed at least cpm_threshold in frac_cols columns
  mae[
    rowSums(cpm(mae[[assay]]) > min_cpm) >
      ncol(mae[[assay]]) * frac_cols,
    ,
  ]
}

# Differential expression helper functions:
de_by_ctrs <- function(df,
                       ctrs,
                       adj_meth = ctrl$adj_meth) {
  lapply(
    ctrs,
    fit = df,
    adj_meth = adj_meth,
    FUN = function(x, fit, adj_meth) {
      y <- DESeq2::results(fit, contrast = x, pAdjustMethod = adj_meth)
      y$feature <- rownames(y)
      y$log_padj <- -1 * log10(y$padj)
      y$ctrs <- paste(x["test"], "vs", x["ref"])
      as.data.frame(y)
    }
  )
}

export_xl <- function(..., info, dir, basenm) {
  
  df_ls <- list(...)
  wb <- openxlsx::createWorkbook()
  
  ws_nm_info <- "info"
  openxlsx::addWorksheet(wb, ws_nm_info)
  info_df <- c(
    info,
    Sys.info(),
    Sys.time()
  )
  info_df <- as.data.frame(info_df)
  openxlsx::writeData(
    wb = wb,
    sheet = ws_nm_info,
    x = info_df,
    rowNames = TRUE,
    colNames = FALSE
  )
  
  for (i in seq_along((df_ls))) {
    df <- df_ls[[i]]
    ws_nm_info <- names(df_ls)[i]
    lbl_df <- lapply(df, attr, "label")
    lbl_df <- lapply(lbl_df, \(x) if (is.null(x)) "" else x)
    lbl_df <- as.data.frame(lbl_df, row.names = "label:")
    openxlsx::addWorksheet(wb, ws_nm_info)
    openxlsx::writeData(
      wb = wb,
      sheet = ws_nm_info,
      x = lbl_df,
      rowNames = TRUE,
      colNames = FALSE
    )
    openxlsx::writeData(
      wb, ws_nm_info,
      x = df,
      rowNames = TRUE,
      startRow = 2
    )
  }
  
  openxlsx::saveWorkbook(
    wb,
    file = file.path(dir, paste0(basenm, ".xlsx")),
    overwrite = TRUE
  )
}
```

#### Data preparation

```{r, ads}
#| code-fold: show
adsl <- dbs.data::adsl
advs <- dbs.data::advs
adlb <- dbs.data::adlb

#' Subjid and Visit to Sample
#' 
subjvis_to_spl <- function(df) paste0(df$subjid, "v", df$avisitn)

ads <- adlb |>
  rbind(advs) |>
  subset(select = -c(ct, dtype, param, base, basetype, chg, pchg)) |>
  tidyr::pivot_wider(names_from = "paramcd", values_from = "aval") |>
  merge(adsl, y = _, by = "subjid") |>
  (\(df) S4Vectors::DataFrame(df, row.names = subjvis_to_spl(df)))() |>
  subset(select = c(subjid, diabcd, diab, avisitn, avisit)) |>
  (\(df) {
    assertthat::assert_that(all(table(subjvis_to_spl(df)) == 1))
    df
  })()

rm(adsl, advs, adlb)
ads
```

```{r, rna}
rna <- list(
  mrna = dbs.data::mrna_raw,
  premirna = dbs.data::premirna_raw,
  mirna = dbs.data::mirna_raw
)

rna[c("premirna", "mirna")] <- lapply(
  X = rna[c("premirna", "mirna")],
  FUN = format_mirna
)

# Rows represent genes
rna <- lapply(X = rna, FUN = function(x) y <- x[rowSums(x) > 0, ])
rna <- lapply(X = rna, as.matrix)
assertthat::assert_that(all(colnames(rna$premirna) == colnames(rna$mirna)))
rna$allmirna <- rbind(rna$premirna, rna$mirna)

library(testthat)
test_that("rna features discriminated in noexpr, expr", {
  lapply(X = rna, FUN = \(x) expect_true(all(rowSums(x) > 0)))
})
```

```{r, mae}
library(MultiAssayExperiment)

#' (Sample-)Map Arrays
#'
#' Use the column names of `x` to deduce the `primary` and `colnames`.
#' This is used to generate the sample mapping between colData and Experiments.
#'
#' @param x (`dataframe`).
#'
#' @note In our case, primary and colnames are equivalent, colnames could
#' be different from primary names when a biological sample has different
#' names in the biological assays (e.g. machine constraint, technical
#' repetitions).
#'
#' @seealso [MultiAssayExperiment::listToMap()]
#' @examples
#' \dontrun{
#'   lapply(rna, map_arrays)
#'   MultiAssayExperiment::listToMap(lapply(rna, map_arrays))
#' }
#'
map_arrays <- function(x) {
  y <- data.frame(colname = colnames(x))
  y$primary <- y$colname
  y
}

besd_mae <- MultiAssayExperiment(
  experiments = ExperimentList(rna),
  colData = ads,
  sampleMap = listToMap(lapply(rna, map_arrays))
)

rm(rna, ads)
besd_mae
```

## Sample depth threshold determination

A scatter plot representing sample gene diversity in relationship with sample
depth evidenced a threshold for miRNA reads at
`r formatC(cfg_prog$rna$depth_threshold$allmirna, format = "e", digits = 0)`
reads for which approximately a million read increase was associated with more
than a 2-fold increase in gene diversity. The depth threshold for mRNA was
fixed at 
`r formatC(cfg_prog$rna$depth_threshold$allmrna, format = "e", digits = 0)`,
although with a less important but still substantial drop in gene variety
below that threshold.

### miRNA

```{r, depth_diversity_mirna}
assay <- "allmirna"
```

::: panel-tabset
#### Static figure

```{r, depth_diversity_gg, eval=cfg_prog$knitr$results!="hide"}
vline <- cfg_prog$rna$depth_threshold[[assay]]
dtaplot <- assays(besd_mae)[[assay]]
dtaplot <- data.frame(
  n_gene = colSums(dtaplot > 0),
  depth  = colSums(dtaplot)
)

dtaplot <- cbind(
  dtaplot,
  data.frame(colData(besd_mae)[rownames(dtaplot),])
)

gg <- ggplot(
  data = dtaplot,
  mapping = aes(x = depth, y = n_gene, colour = diab)
) +
  geom_point() +
  geom_vline(xintercept = vline) +
  xlab("Sample depth") +
  ylab("Gene diversity (number of genes with reads)") +
  coord_cartesian(ylim = c(0, NA)) +
  theme_minimal() +
  theme(legend.position = "top")

gg
```

#### Interactive figure

```{r, depth_diversity_plotly, eval=cfg_prog$knitr$results!="hide"}
plotly::ggplotly(gg)
```
:::

```{r}
rm(vline, dtaplot, assay, gg)
```

### mRNA

::: panel-tabset
#### Static figure

```{r, depth_diversity_mrna}
assay <- "mrna"
```

```{r, depth_diversity_gg, eval=cfg_prog$knitr$results!="hide"}
```

#### Interactive figure

```{r, depth_diversity_plotly, eval=cfg_prog$knitr$results!="hide"}
```
:::

:::

## `dds_1` - DE: Baseline, all micro RNA

```{r, dds_1_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "allmirna"

dds_1_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(
    assay = ngs_assay,
    min_depth = ctrl$depth_threshold[[ngs_assay]]
  ) |>
  filter_for_coldata(col = "avisit", is = "Baseline") |>
  filter_for_low_expr(
    assay = ngs_assay,
    min_cpm = ctrl$cpm_threshold[[ngs_assay]],
    frac_cols = ctrl$cpm_threshold$fraccol
  ) |>
  (\(mae) DESeq2::DESeqDataSetFromMatrix(
    countData = mae[[ngs_assay]],
    colData = colData(x = mae),
    design = stats::formula(~ diabcd)
  ) )() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam(),
    fitType = "local"
  )
```

```{r, dds_1_est}
dds_1_est <- dds_1_fit |>
  de_by_ctrs(
    ctrs = list(
      c("diabcd", test = "T2D", ref = "NGT"),
      c("diabcd", test = "T2D", ref = "IGT"),
      c("diabcd", test = "IGT", ref = "NGT")
    )
  ) |>
  Reduce(rbind, x = _)
```

### Fig 07 01

```{r, dds_1_gg}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_1_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)
dds_1_gg <-
  dds_1_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )
```

```{r, dds_1_an}
p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.1",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all miRNA at Baseline", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_1_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )
```

```{r, dds_1_xp}
export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_01.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_01_af.pdf")
)
```

```{r, f07_01_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_01_xlsx}
export_xl(
  fig_07_01 = dds_1_est,
  info = c(
    title =
      "miRNA at baseline - differential expression by diabetes group",
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_01"
)
```

### Fig 07 02

```{r, f07_02}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_1_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_1_est$label <- gsub(
  "hsa-(mir|miR|let)-",
  dds_1_est$feature,
  replacement = ""
)
dds_1_est$type <- substr(dds_1_est$feature, start = 5, stop = 7)
head(dds_1_est)

dds_1_gg <- ggplot(
  dds_1_est,
  mapping = aes(x = log2FoldChange, y = log_padj, col = type)
) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(
    data = subset(dds_1_est, padj < 0.001 & abs(log2FoldChange) > 1),
    size = 1,
    shape = 3
  ) +
  geom_text(
    data = subset(dds_1_est, padj < 0.001 & abs(log2FoldChange) > 1),
    aes(label = label),
    vjust = -0.5,
    size = 2,
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.2",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all miRNA at Baseline", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_1_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_02.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_02_af.pdf")
)

```

```{r, f07_02_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, clean_dds_1}
rm(ctrl, ngs_assay, dds_1_fit, pval, log_adj, brk, dds_1_gg, p)
gc()
ls()
```

## `dds_2` - DE: Baseline, mRNA

```{r, dds_2_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "mrna"

dds_2_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(
    assay = ngs_assay,
    min_depth = ctrl$depth_threshold[[ngs_assay]]
  ) |>
  filter_for_coldata(col = "avisit", is = "Baseline") |>
  filter_for_low_expr(
    assay = ngs_assay,
    min_cpm = ctrl$cpm_threshold[[ngs_assay]]
  ) |>
  (\(mae) DESeq2::DESeqDataSetFromMatrix(
    countData = mae[[ngs_assay]],
    colData = colData(x = mae),
    design = stats::formula(~ diabcd)
  ) )() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam()
  )
```

```{r, dds_2_est}
dds_2_est <- dds_2_fit |>
  de_by_ctrs(
    ctrs = list(
      c("diabcd", test = "T2D", ref = "NGT"),
      c("diabcd", test = "T2D", ref = "IGT"),
      c("diabcd", test = "IGT", ref = "NGT")
    )
  ) |>
  Reduce(rbind, x = _)
```

```{r, eval=FALSE}
head(dds_2_est)


ndegenes <- dds_2_est |>
  subset(round(padj, 5) < 0.001) |>
  with(unique(feature)) |>
  length()


sprintf("%.2f", round(ndegenes / ngenes * 100, 2))

split(dds_2_est, dds_2_est$ctrs) |>
  lapply(\(x) {
    dim(x)
    head(x)
    table(cut(x$padj, breaks = c(0, 0.001, 0.05, 1)))
  })



pct_de_genes <- sprintf("%.2f", round(ndegenes / ngenes * 100, 2))


genes_nsub <- nrow(besd_mae[[ngs_assay]])
genes_de_n <- c(
  Overall = dds_2_est |>
      subset(round(padj, 5) < 0.001) |>
      with(unique(feature)) |>
      length()
  ,
  split(dds_2_est, dds_2_est$ctrs) |>
    sapply(\(x) {
      x <- subset(x, round(padj, 5) < 0.001)
      length(unique(x$feature))
    })
)

genes_nsub
genes_de_n
genes_de_pct <- sprintf("%.2f", genes_de_n / genes_nsub * 100)

```

### Fig 07 03

```{r, dds_2_gg}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_2_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_2_gg <-
  dds_2_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )
```

```{r, dds_2_an}
p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.3",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among mRNA at Baseline", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_2_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )
```

```{r, dds_2_xp}
export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_03.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_03_af.pdf")
)
```

```{r, f07_03_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_03_xlsx}
export_xl(
  fig_07_03 = dds_2_est,
  info = c(
    title =
      "mRNA at baseline - differential expression by diabetes group",
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_03"
)
```

### Fig 07 04

```{r, f07_04}
dds_2_est$label <- gsub("ENSG0+", "", dds_2_est$feature)
head(dds_2_est)

pval <- c(0.05, 0.001)
log_adj <- pretty(dds_2_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_2_gg <- ggplot(dds_2_est, mapping = aes(x = log2FoldChange, y = log_padj)) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(
    data = subset(dds_2_est, padj < 0.001 & abs(log2FoldChange) > 2),
    size = 1,
    shape = 3,
    color = "royalblue"
  ) +
  geom_text(
    data = subset(dds_2_est, padj < 0.001 & abs(log2FoldChange) > 2),
    aes(label = label),
    vjust = -0.5,
    size = 2,
    color = "royalblue",
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.4",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all miRNA at Baseline", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_2_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_04.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_04_af.pdf")
)

```

```{r, f07_04_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

## `dds_3` - DE: Month 3, all micro RNA

```{r, dds_3_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "allmirna"

dds_3_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(
    assay = ngs_assay,
    min_depth = ctrl$depth_threshold[[ngs_assay]]
  ) |>
  filter_for_coldata(col = "avisit", is = "Month 3") |>
  filter_for_low_expr(
    assay = ngs_assay,
    min_cpm = ctrl$cpm_threshold[[ngs_assay]]
  ) |>
  (\(mae) DESeq2::DESeqDataSetFromMatrix(
    countData = mae[[ngs_assay]],
    colData = colData(x = mae),
    design = stats::formula(~ diabcd)
  ) )() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam(),
    fitType = "local"
  )
```

```{r, dds_3_est}
dds_3_est <-
  dds_3_fit |>
  de_by_ctrs(
    ctrs = list(
      c("diabcd", test = "T2D", ref = "NGT"),
      c("diabcd", test = "T2D", ref = "IGT"),
      c("diabcd", test = "IGT", ref = "NGT")
    )
  ) |>
  Reduce(rbind, x = _)
```

### Fig 07 05

```{r, f07_05}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_3_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)
dds_3_gg <-
  dds_3_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.5",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all miRNA at Month 3", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_3_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_05.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_05_af.pdf")
)
```

```{r, f07_05_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_05_xlsx}
export_xl(
  fig_07_05 = dds_3_est,
  info = c(
    title =
      "All miRNA at month 3 - differential expression by diabetes group",
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_05"
)
```

### Fig 07 06

```{r, f07_06}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_3_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_3_est$label <- gsub(
  "hsa-(mir|miR|let)-",
  dds_3_est$feature,
  replacement = ""
)
dds_3_est$type <- substr(dds_3_est$feature, start = 5, stop = 7)
head(dds_3_est)

dds_3_gg <- ggplot(
  dds_3_est,
  mapping = aes(x = log2FoldChange, y = log_padj, col = type)
) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(
    data = subset(dds_3_est, padj < 0.001 & abs(log2FoldChange) > 1),
    size = 1,
    shape = 3
  ) +
  geom_text(
    data = subset(dds_3_est, padj < 0.001 & abs(log2FoldChange) > 1),
    aes(label = label),
    vjust = -0.5,
    size = 2,
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.6",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all miRNA at Month 3", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_3_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_06.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_06_af.pdf")
)

```

```{r, f07_06_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

## `dds_4` - DE: Month 3, mRNA

```{r, dds_4_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "mrna"

dds_4_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(
    assay = ngs_assay,
    min_depth = ctrl$depth_threshold[[ngs_assay]]
  ) |>
  filter_for_coldata(col = "avisit", is = "Month 3") |>
  filter_for_low_expr(
    assay = ngs_assay,
    min_cpm = ctrl$cpm_threshold[[ngs_assay]]
  ) |>
  (\(mae) DESeq2::DESeqDataSetFromMatrix(
    countData = mae[[ngs_assay]],
    colData = colData(x = mae),
    design = stats::formula(~ diabcd)
  ) )() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam()
  )
```

```{r, dds_4_est}
dds_4_est <-
  dds_4_fit |>
  de_by_ctrs(
    ctrs = list(
      c("diabcd", test = "T2D", ref = "NGT"),
      c("diabcd", test = "T2D", ref = "IGT"),
      c("diabcd", test = "IGT", ref = "NGT")
    )
  ) |>
  Reduce(rbind, x = _)
```

### Fig 07 07

```{r, f07_07}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_4_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_4_gg <-
  dds_4_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.7",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among all mRNA at Month 3", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_4_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_07.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_07_af.pdf")
)
```

```{r, f07_07_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_07_xlsx}
export_xl(
  fig_07_07 = dds_4_est,
  info = c(
    title =
      "mRNA at month 3 - differential expression by diabetes group",
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_07"
)
```

### Fig 07 08

```{r, f07_08}
dds_4_est$label <- gsub("ENSG0+", "", dds_4_est$feature)
head(dds_4_est)

pval <- c(0.05, 0.001)
log_adj <- pretty(dds_4_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_4_gg <- ggplot(dds_4_est, mapping = aes(x = log2FoldChange, y = log_padj)) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(
    data = subset(dds_4_est, padj < 0.001 & abs(log2FoldChange) > 2),
    size = 1,
    shape = 3,
    color = "royalblue"
  ) +
  geom_text(
    data = subset(dds_4_est, padj < 0.001 & abs(log2FoldChange) > 2),
    aes(label = label),
    vjust = -0.5,
    size = 2,
    color = "royalblue",
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.8",
      strwrap(
        "Volcano plot - Response Size and Significance of Differential
        Expression among mRNA at Month 3", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_4_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_08.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_08_af.pdf")
)

```

```{r, f07_08_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

## `dds_5` - DE: Exercise intervention, all micro RNA

```{r, dds_5_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "allmirna"

filter_for_paired_visits <- function(mae) {
  assert_that(all(table(colData(mae)$subjid) <= 2))
  
  subj <- table(colData(mae)$subjid)
  subj <- names(subj[subj == 2])
  mae <- mae[, colData(mae)$subjid %in% subj, ]
  
  chk_df <- as.data.frame(colData(mae))
  check <- split(chk_df, f = chk_df$avisit)
  
  # Verify the selection:
  assert_that(all(
    sapply(check, subj = subj,\(x, subj) all(subj %in% x$subjid))
  ))
  mae
}

dds_5_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(ngs_assay, ctrl$depth_threshold[[ngs_assay]]) |>
  filter_for_paired_visits() |>
  filter_for_low_expr(ngs_assay, ctrl$cpm_threshold[[ngs_assay]]) |>
  (\(mae) {
    # <https://support.bioconductor.org/p/84241/#84296>
    # Or:
    # vignette('DESeq2'),
    # > Section: "Group-specific condition effects, individuals nested
    # > within groups"
    colData(mae) <- droplevels(colData(mae))
    colData(mae)$subjid <- as.character(colData(mae)$subjid)
    colData(mae)$avisit_ <- factor(
      colData(mae)$avisit,
      levels = levels(colData(mae)$avisit),
      labels = sub(" ", "_", levels(colData(mae)$avisit))
    )
    coldata <- colData(mae)[c("diabcd", "subjid", "avisit_")]
    
    coldata <- do.call(
      rbind,
      by(
        data = coldata,
        INDICES = list(coldata$diabcd),
        FUN = function(x) {
          subjid <- as.factor(x$subjid)# nolint
          x$P.n <- factor(# nolint
            subjid,
            levels = levels(subjid),
            labels = paste0("P.", 1:nlevels(subjid))
          )
          x[order(x$subjid, x$diabcd), ]
        }
      )
    )
    
    coldata <- coldata[rownames(colData(mae)), ]
    dsgn <- stats::model.matrix(
      ~diabcd + diabcd:P.n + diabcd:avisit_,
      coldata
    )
    dsgn <- dsgn[, colSums(dsgn) != 0]
    
    DESeq2::DESeqDataSetFromMatrix(
      countData = mae[[ngs_assay]],
      colData = colData(x = mae),
      design = dsgn
    )
  })() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam(),
    fitType = "local",
    betaPrior = FALSE
  )

de_by_name <- function(df,
                       name,
                       adj_meth = ctrl$adj_meth) {
  lapply(
    name,
    fit = df,
    adj_meth = adj_meth,
    FUN = function(x, fit, adj_meth) {
      y <- DESeq2::results(fit, name = x, pAdjustMethod = adj_meth)
      y$feature <- rownames(y)
      y$log_padj <- -1 * log10(y$padj)
      y$ctrs <- gsub(
        pattern = "^diabcd(...)\\.avisit.*$",
        x = x,
        replacement = "chg \\1"
      )
      as.data.frame(y)
    }
  )
}

dds_5_est <-
  dds_5_fit |>
  de_by_name(
    name = c(
      chg_ngt = "diabcdNGT.avisit_Month_3",
      chg_igt = "diabcdIGT.avisit_Month_3",
      chg_t2d = "diabcdT2D.avisit_Month_3"
    )
  ) |>
  Reduce(rbind, x = _) |>
  within(
    ctrs <- factor(ctrs, levels = paste("chg", c("NGT", "IGT", "T2D")))
  )
```

### Fig 07 09

```{r, f07_09}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_5_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_5_gg <-
  dds_5_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.9",
      strwrap(
        "Volcano plot - Size and Significance of Differential
        Expression among micro RNA in response to the exercise
        intervention (CHG)", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_5_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_09.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_09_af.pdf")
)
```

```{r, f07_09_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_09_xlsx}
export_xl(
  fig_07_09 = dds_5_est,
  info = c(
    title = paste(
      "All miRNA Change from Baseline - differential expression by diabetes",
      "group"
    ),
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_09"
)
```

### Fig 07 10

```{r, f07_10}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_5_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_5_est$label <- gsub(
  "hsa-(mir|miR|let)-",
  dds_5_est$feature,
  replacement = ""
)
dds_5_est$type <- substr(dds_5_est$feature, start = 5, stop = 7)
head(dds_5_est)

dds_5_txt <- subset(dds_5_est, padj < 0.05 & abs(log2FoldChange) > .5)

dds_5_gg <- ggplot(
  dds_5_est,
  mapping = aes(x = log2FoldChange, y = log_padj, col = type)
) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(data = dds_5_txt, size = 1, shape = 3) +
  geom_text(
    data = dds_5_txt,
    aes(label = label),
    vjust = -0.5,
    size = 2,
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.10",
      strwrap(
        "Volcano plot - Size and Significance of Differential
        Expression among micro RNA in response to the exercise
        intervention (CHG)", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_5_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_10.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_10_af.pdf")
)
```

```{r, f07_10_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

## `dds_6` - DE: Exercise intervention, all micro RNA

```{r, dds_6_dta}
ctrl <- cfg_prog$rna
ngs_assay <- "mrna"

filter_for_paired_visits <- function(mae) {
  assert_that(all(table(colData(mae)$subjid) <= 2))
  
  subj <- table(colData(mae)$subjid)
  subj <- names(subj[subj == 2])
  mae <- mae[, colData(mae)$subjid %in% subj, ]
  
  chk_df <- as.data.frame(colData(mae))
  check <- split(chk_df, f = chk_df$avisit)
  
  # Verify the selection:
  assert_that(all(
    sapply(check, subj = subj,\(x, subj) all(subj %in% x$subjid))
  ))
  mae
}

dds_6_fit <- besd_mae |>
  (\(mae) mae[ , , ngs_assay])() |>
  filter_for_depth(ngs_assay, ctrl$depth_threshold[[ngs_assay]]) |>
  filter_for_paired_visits() |>
  filter_for_low_expr(ngs_assay, ctrl$cpm_threshold[[ngs_assay]]) |>
  (\(mae) {
    # <https://support.bioconductor.org/p/84241/#84296>
    # Or:
    # vignette('DESeq2'),
    # > Section: "Group-specific condition effects, individuals nested
    # > within groups"
    colData(mae) <- droplevels(colData(mae))
    colData(mae)$subjid <- as.character(colData(mae)$subjid)
    colData(mae)$avisit_ <- factor(
      colData(mae)$avisit,
      levels = levels(colData(mae)$avisit),
      labels = sub(" ", "_", levels(colData(mae)$avisit))
    )
    coldata <- colData(mae)[c("diabcd", "subjid", "avisit_")]
    
    coldata <- do.call(
      rbind,
      by(
        data = coldata,
        INDICES = list(coldata$diabcd),
        FUN = function(x) {
          subjid <- as.factor(x$subjid)# nolint
          x$P.n <- factor(# nolint
            subjid,
            levels = levels(subjid),
            labels = paste0("P.", 1:nlevels(subjid))
          )
          x[order(x$subjid, x$diabcd), ]
        }
      )
    )
    
    coldata <- coldata[rownames(colData(mae)), ]
    dsgn <- stats::model.matrix(
      ~diabcd + diabcd:P.n + diabcd:avisit_,
      coldata
    )
    dsgn <- dsgn[, colSums(dsgn) != 0]
    
    DESeq2::DESeqDataSetFromMatrix(
      countData = mae[[ngs_assay]],
      colData = colData(x = mae),
      design = dsgn
    )
  })() |>
  DESeq2::DESeq(
    parallel = TRUE,
    BPPARAM = BiocParallel::bpparam(),
    fitType = "local",
    betaPrior = FALSE
  )

dds_6_est <-
  dds_6_fit |>
  de_by_name(
    name = c(
      chg_ngt = "diabcdNGT.avisit_Month_3",
      chg_igt = "diabcdIGT.avisit_Month_3",
      chg_t2d = "diabcdT2D.avisit_Month_3"
    )
  ) |>
  Reduce(rbind, x = _) |>
  within(
    ctrs <- factor(ctrs, levels = paste("chg", c("NGT", "IGT", "T2D")))
  )
```

### Fig 07 11

```{r, f07_11}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_6_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_6_gg <-
  dds_6_est |>
  ggplot(aes(log2FoldChange, log_padj, fill = log10(..count..))) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex() +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_y_continuous(breaks = brk) +
  facet_grid(. ~ ctrs) +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(3, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.11",
      strwrap(
        "Volcano plot - Size and Significance of Differential
        Expression among mRNA in response to the exercise
        intervention (CHG)", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_6_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_11.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_11_af.pdf")
)
```

```{r, f07_11_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, f07_11_xlsx}
export_xl(
  fig_07_11 = dds_6_est,
  info = c(
    title = paste(
      "mRNA Change from Baseline - differential expression by diabetes",
      "group"
    ),
    source = "UMB-BESD"
  ),
  dir = cfg_prog$paths$grh,
  basenm = "fig_07_11"
)
```

### Fig 07 12

```{r, f07_12}
pval <- c(0.05, 0.001)
log_adj <- pretty(dds_6_est$log_padj)
brk <- setNames(
  c(log_adj, -1 * log10(pval)),
  c(log_adj, paste0("padj=", pval))
)

dds_6_est$label <- gsub(
  "hsa-(mir|miR|let)-",
  dds_6_est$feature,
  replacement = ""
)
dds_6_est$type <- substr(dds_6_est$feature, start = 5, stop = 7)
head(dds_6_est)

dds_6_txt <- subset(dds_6_est, padj < 0.05 & abs(log2FoldChange) > .5)

dds_6_gg <- ggplot(
  dds_6_est,
  mapping = aes(x = log2FoldChange, y = log_padj, col = type)
) +
  geom_vline(xintercept = c(-1, 1), lty = 2) +
  xlab("Log2-fold-change") +
  ylab(expression(-1 %*% log10(padj))) +
  stat_bin_hex(mapping = aes(fill = log10(..count..)), color = "transparent") +
  geom_point(data = dds_6_txt, size = 1, shape = 3) +
  geom_text(
    data = dds_6_txt,
    aes(label = label),
    vjust = -0.5,
    size = 2,
    show.legend = FALSE
  ) +
  scale_fill_gradient(low = "black", high = "gray90") +
  scale_color_manual(values = c("#FF5003", "#003F7D")) +
  scale_y_continuous(breaks = brk) +
  facet_wrap(. ~ ctrs, ncol = 3) +
  coord_cartesian(clip = "off") +
  theme_minimal() +
  theme(
    text = element_text(size = 7),
    title =  element_text(size = 7),
    legend.position = "bottom",
    legend.key.height = unit(.5, "lines"),
    legend.key.width = unit(2, "lines"),
    legend.text.align = 0,
    panel.grid.minor = element_blank()
  )

p <- clean_slate() |>
  add_header(c("FCA Collin", "UMB BESD"), c("Confidential", "Draft")) |>
  add_title(
    c(
      "Figure 7.12",
      strwrap(
        "Volcano plot - Size and Significance of Differential
        Expression among mRNA in response to the exercise
        intervention (CHG)", width = 80
      ),
      "Analysis Set: Full Analysis Set"
    )
  ) |>
  add_figure(dds_6_gg, height = inches(3), width = inches(6)) |>
  add_footer(
    c(
      "Program t2d_07_mir / Env ayup_dbs:v0.1.0-alpha",
      format(Sys.time(), format = "%Y-%m-%d %H:%M (%Z)")
    ),
    cfg_prog$version
  )

export_as(
  p,
  file = file.path(cfg_prog$paths$grh, "fig_07_12.pdf"),
  file_graph_alone = file.path(cfg_prog$paths$grh, "fig_07_12_af.pdf")
)
```

```{r, f07_12_sh, eval=cfg_prog$knitr$results!="hide"}
#| fig-width: 8.3
#| fig-height: 11.7
show_slate(p)
```

```{r, clean_dds_6}
rm(ctrl, ngs_assay, dds_6_fit, pval, log_adj, brk, dds_6_gg, p)
gc()
ls()
```

## Session Informations

```{r}
sessioninfo::session_info()
```
